src/main.rs

```diff
@@ -1,9 +1,23 @@
 extern crate openzwave_stateful as openzwave;
 use openzwave::InitOptions;
-use openzwave::{ ValueGenre, ValueID };
+use openzwave::{ ValueGenre, ValueID, ZWaveNotification };
 
-use std::io;
+use std::{ io, thread };
 use std::io::Write;
+use std::sync::mpsc;
+
+fn display_prompt() {
+    print!("> ");
+    io::stdout().flush().unwrap(); // https://github.com/rust-lang/rust/issues/23818
+}
+
+fn spawn_notification_thread(rx: mpsc::Receiver<ZWaveNotification>) {
+    thread::spawn(move || {
+        for notification in rx {
+            println!("{}", notification);
+        }
+    });
```

Comment:
The problem is that we can't both read_line on the input, and listen for the notification, as each of these actions are blocking. I think one of those actions definitely need to be in a separate thread (and even possibly each in its own thread to keep the main thread free ?).

In the adapter I don't think I'll need to spawn a thread though, as I think a thread is already spawned by the adapter manager.